### PR TITLE
add a test for audio jitter in DEV/PR builds

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

**Comment by zzmp** [comment]
QA:

There is a new menu item, under Settings > Audio... > Debug gate threshold.
This should start at 1. You can toggle it, and it will throttle the network's audio packets so that they are processed in groups of size N, where you set N.
What should happen:
- If you increase it by a lot, you will hear silence as the jitter buffer fills, and then it should be smooth audio again. You will see logs saying that it increased the desired jitter buffer frames.
- If you decrease it by a lot, intermediate frames will be dropped. You may hear it. You will see logs saying that the frames were dropped. Eventually, the desired jitter buffer frames will decrease again.
- The setting should stay across sessions.

This is a debugging tool, explicitly. It will not appear in release builds.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3415d4892459b3def83e6e107b21b3dc8118009c available at: https://gist.github.com/b92d6f864421d27ddef0ecc1a64ab40b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8588-3415d48.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8588-3415d48.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e785c0cdf801bffd2a007e53ca1e716921749eee available at: https://gist.github.com/b92d6f864421d27ddef0ecc1a64ab40b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8588-e785c0c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8588-e785c0c.dmg

File: libraries/audio-client/src/AudioClient.cpp [hunk 1]
```diff
@@ -99,6 +101,8 @@ class CheckDevicesThread : public QThread {
 
 AudioClient::AudioClient() :
     AbstractAudioInterface(),
+    _gateThreshold("audioOutputGateThreshold", DEFAULT_AUDIO_OUTPUT_GATE_THRESHOLD),
```
Comment (ZappoMan) [inline libraries/audio-client/src/AudioClient.cpp]: shouldn't all of these also be inside the `#if DEV_BUILD || PR_BUILD` wrapper?

File: libraries/audio-client/src/AudioClient.cpp [hunk 2]
```diff
@@ -540,11 +544,48 @@ void AudioClient::handleAudioDataPacket(QSharedPointer<ReceivedMessage> message)
...
+#if DEV_BUILD || PR_BUILD
+        _gate.insert(message);
+#else
         // Audio output must exist and be correctly set up if we're going to process received audio
         _receivedAudioStream.parseData(*message);
+#endif
     }
 }
 
+AudioClient::Gate::Gate(AudioClient* audioClient, int threshold) :
```
Comment (ZappoMan) [inline libraries/audio-client/src/AudioClient.cpp]: shouldn't all of these also be inside the `#if DEV_BUILD || PR_BUILD` wrapper?

**Comment by zzmp** [comment]
@ZappoMan I fear that by putting all of the AudioClient::Gate logic inside of `#if DEV_BUILD || PR_BUILD` guards, I will miss one. I do not think Jenkins will catch it (it will be doing a `PR_BUILD`), and it will make the code less maintainable.

It doesn't hurt performance to have the AudioClient::Gate, so long as it is unused, so I just left it in with a single guard where to activate it in those builds, and a single guard to expose it to the settings in those builds.

**Comment by ZappoMan** [comment]
@zzmp ok

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: e785c0cdf801bffd2a007e53ca1e716921749eee available at: https://gist.github.com/b92d6f864421d27ddef0ecc1a64ab40b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8588-e785c0c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8588-e785c0c.dmg

**Comment by ZappoMan** [comment]
Tested: Approved

**Pull request merged**